azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AddMojo.java

```diff
@@ -319,6 +341,40 @@ protected void preparePackageName() throws MojoFailureException {
         return params;
     }
 
+    protected String getStringInputFromUser(String attributeName, String initValue, FunctionSettingTemplate template) {
+        final String regex = template == null ? null : template.getSettingRegex();
+        final String defaultValue = template == null ? null : template.getDefaultValue();
+        final String prompt = defaultValue == null ?
+            String.format("Enter value for %s: ", attributeName) :
+            String.format("Enter value for %s(Default: %s): ", attributeName, defaultValue);
+        final String errorMessage = (template != null && template.getErrorText() != null) ?
+            TemplateResources.getResource(template.getErrorText()) : DEFAULT_INPUT_ERROR_MESSAGE;
+        final Function<String, Boolean> validator = regex == null ?
```

Comment:
Too much `(?:)` in this method, which makes the logic here a little bit confusing.

You can write it like this:

```java 
Function<String, Boolean> validator = StringUtils::isNotEmpty;
if (template != null && StringUtils.isNotEmpty(template.getSettingRegex())) {
    validator = (attribute) -> StringUtils.isNotEmpty(attribute) && attribute.matches(template.getSettingRegex());
}
```

Reply:
Done, and move most of them to simple function